backend/fms_core/tests/test_template_importers/project_link_samples_test.py

```diff
@@ -75,12 +75,11 @@ def test_import(self):
         self.assertEqual(result['valid'], True)
 
         #Custom tests for each template
-        self.assertEqual(len(SampleByProject.objects.all()), 3)
-        self.assertTrue(SampleByProject.objects.filter(sample=self.sample1, project=self.project1).exists())
-        self.assertTrue(SampleByProject.objects.filter(sample=self.sample2, project=self.project1).exists())
-        self.assertTrue(SampleByProject.objects.filter(sample=self.sample2, project=self.project2).exists())
-        self.assertFalse(SampleByProject.objects.filter(sample=self.sample1, project=self.project2).exists())
-        self.assertFalse(SampleByProject.objects.filter(sample=self.sample3, project=self.project3).exists())
+        self.assertEqual(len(DerivedSample.objects.filter(project__isnull=False).all()), 2)
+        self.assertTrue(DerivedSample.objects.filter(samples__in=self.sample1, project=self.project3).exists())
```

Comment:
Change samples__in look up to just sample since it is not an array. Same for the next 3 lines.

Reply:
samples is the reverse relation of derived_samples... its likely to hold more than 1 sample. sample does not exist on DerivedSample... I'll test for "=" instead of "__in".